src/typing/fields.ml

```diff
@@ -6,6 +6,29 @@ open TyperBase
 open Error
 open Typecore
 
+module TypeFieldConfig = struct
+	type t = {
+		allow_resolve : bool;
+		do_resume : bool;
+	}
+
+	let allow_resolve cfg = cfg.allow_resolve
+
+	let do_resume cfg = cfg.do_resume
+
+	let default = {
+		allow_resolve = true;
+		do_resume = false;
+	}
+
+	let create resume = {
```

Comment:
Maybe this function is not needed since half of the time you create `TypeFieldConfig.t` bypassing it. And it's almost the same amount of code in `TypeFieldConfig.create resume` and `{do_resume = resume;allow_resolve = true}`

Reply:
Yes but if we add another configuration we would have to change all these places.

Reply:
It could happen, we'll need to change all occurances of `TypeFieldConfig.create resume` too if the new parameter will have different values in those places. 
Who knows :) But in the current state of the code this function seems to be unnecessary as it suggests that it to be always used to create `t`.
I'm fine with it anyway.

Reply:
Well, ideally we would force creation through this function, but I don't really know how to do that.

Reply:
Something like this should work:
```ocaml
module type TypeFieldConfigInterface = sig
	type t
	val allow_resolve : t -> bool
	val do_resume : t -> bool
	val default : t
	val create : bool -> t
	val with_resume : t -> t

end

module TypeFieldConfig : TypeFieldConfigInterface = struct
<...>
```

Reply:
Nice, I didn't realize you could apply module types like that. I might use that in some places to improve encapsulation. Thanks!

Reply:
Maybe someday we won't be a bunch of clowns making flash games in the eyes of OCaml people :-3

Reply:
You mean when Haxe has been ported to Haxe? :D